### ISSUE TITLE
Deleting a cluster errors

### ISSUE THREAD

**Issue by worldofgeese** [syseleven/terraform-provider-metakube#46]
I needed to change the the `subnet_cidr` of my cluster. On `apply` terraform reasonably printed the necessary destruction/creation lifecycle to change the subnet. Node pools successfully destroyed but the cluster appears to be stuck destroying itself and terraform errored with 

```
metakube_cluster.this: Still destroying... [id=ztdlsqv4hq, 17m0s elapsed]
metakube_cluster.this: Still destroying... [id=ztdlsqv4hq, 17m10s elapsed]
metakube_cluster.this: Still destroying... [id=ztdlsqv4hq, 17m20s elapsed]
╷
│ Error: unable to get cluster 'ztdlsqv4hq': &{<nil>} (*models.ErrorResponse) is not supported by the TextConsumer, can be resolved by supporting TextUnmarshaler interface
```

Assuming this means the cluster was advanced to a state of destruction that the id no longer exists, the terraform provider should probably have a way of dealing with this that isn't aborting the `apply`.

**Comment by furkhat** [comment]
Hi, thanks for opening the issue!

When API returs 404 terraform finishes gracefully. It looks like API returned unexpected reply and terraform stopped. I added basic retry and this should be fixed.

**Comment by furkhat** [comment]
Included in `v1.0.1`
